pkg/util/tradingutil/orders.go

```diff
@@ -0,0 +1,53 @@
+package tradingutil
+
+import (
+	"fmt"
+
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+func HasSufficientBalance(
+	baseCurrency string,
+	quoteCurrency string,
+	side types.SideType,
+	price, quantity fixedpoint.Value,
+	tradingMarket types.Market,
+	balances types.BalanceMap,
+) (bool, string) {
```

Comment:
i think the problem of this function you got too many arguments, better to avoid it

For side buy, you don't need Base asset balance, you only need Quote asset balance and the buy price to calculate the actual quantity.

For side sell, you only need Base asset balance, quote balance and price are not used for this case

Reply:
I want to have a unified way to check the balance when submitting an order with `SubmitOrder`.
Maybe it will be better if I pass `SubmitOrder` instead.

Reply:
Updated function signature:
```
func HasSufficientBalance(
	tradingMarket types.Market,
	submitOrder types.SubmitOrder,
	balances types.BalanceMap,
) (bool, error)
```